app/presenters/dttp/trainee_presenter.rb

```diff
@@ -2,36 +2,55 @@
 
 module Dttp
   class TraineePresenter
-    attr_reader :trainee
+    FEMALE_GENDER_CODE = 1
+    MALE_GENDER_CODE = 2
 
     delegate_missing_to :trainee
 
+    attr_reader :trainee
+
     def initialize(trainee:)
       @trainee = trainee
     end
 
-    def to_dttp_params
+    def contact_params
       {
         "firstname" => trainee.first_names,
         "lastname" => trainee.last_name,
-        "contactid" => trainee.dttp_id,
         "address1_line1" => trainee.address_line_one,
         "address1_line2" => trainee.address_line_two,
         "address1_line3" => trainee.town_city,
         "address1_postalcode" => trainee.postcode,
-        "birthdate" => formatted_dob,
+        "birthdate" => trainee.date_of_birth.to_s,
         "emailaddress1" => trainee.email,
-        "gendercode" => trainee.gender,
+        "gendercode" => gender_code,
         "mobilephone" => trainee.phone_number,
-        "_dfe_ethnicityid_value" => dttp_ethnicity_id,
-        "_dfe_disibilityid_value" => dttp_disability_id,
+        "dfe_EthnicityId@odata.bind" => "/dfe_ethnicities(#{dttp_ethnicity_id})",
+        "dfe_DisibilityId@odata.bind" => "/dfe_disabilities(#{dttp_disability_id})",
+      }
+    end
+
+    def placement_assignment_params
+      {
+        "dfe_programmestartdate" => trainee.programme_start_date.in_time_zone.iso8601,
+        "dfe_ContactId@odata.bind" => "/contacts(#{trainee.dttp_id})",
+        "dfe_CoursePhaseId@odata.bind" => "/dfe_coursephases(#{dttp_course_phase_id})",
+        "dfe_ITTSubject1Id@odata.bind" => "/dfe_subjects(#{dttp_subject_id})",
       }
     end
 
   private
 
-    def formatted_dob
-      trainee.date_of_birth.strftime("%d/%m/%Y")
+    def gender_code
+      trainee.gender == "male" ? MALE_GENDER_CODE : FEMALE_GENDER_CODE
```

Comment:
I think you could use `trainee.male?` here

Reply:
I was thinking the same, but then I thought, what's the point unless it's going to be used in other places. I don't think the rest of the application cares if a trainee is male or female and I'm always weary of adding new behaviour to a model - they have a tendancy to get "fat" over time. This is where presenters shine - they help models avoid unnecessary baggage when it comes to data transformation. That's my 2c on it anyway. If you feel strongly about it, I won't object since it's a very common pattern and it wouldn't be a big change.

Reply:
I very much agree with you but in this scenario, it's available out of the box by Rails via `enum gender: { male: 0, female: 1, other: 2 }` :-). I'm not too fussed over this to be honest, I prefer to keep this sort of stuff out of models too.

Reply:
I take my comment back. I didn't realise enums created these sort of methods - nice!. :)